runtime/src/iree/hal/drivers/local_task/task_driver.c

```diff
@@ -162,13 +162,28 @@ static iree_status_t iree_hal_task_driver_create_device_by_path(
     iree_string_view_t device_path, iree_host_size_t param_count,
     const iree_string_pair_t* params, iree_allocator_t host_allocator,
     iree_hal_device_t** out_device) {
-  if (!iree_string_view_is_empty(device_path)) {
-    return iree_make_status(IREE_STATUS_NOT_FOUND,
-                            "device paths not yet implemented");
+  if (iree_string_view_is_empty(device_path)) {
+    return iree_hal_task_driver_create_device_by_id(
+        base_driver, IREE_HAL_DEVICE_ID_DEFAULT, param_count, params,
+        host_allocator, out_device);
   }
-  return iree_hal_task_driver_create_device_by_id(
-      base_driver, IREE_HAL_DEVICE_ID_DEFAULT, param_count, params,
-      host_allocator, out_device);
+
+  // Try to parse as a device index.
+  uint32_t device_index = 0;
+  if (iree_string_view_atoi_uint32(device_path, &device_index)) {
+    // Right now we only ever expose 1 device. So error out for all other cases.
+    if (device_index != 0) {
+      return iree_make_status(
+          IREE_STATUS_NOT_FOUND,
+          "device index %u invalid; only %u device(s) available", device_index,
+          1);
+    }
+    return iree_hal_task_driver_create_device_by_id(
+        base_driver, IREE_HAL_DEVICE_ID_DEFAULT, param_count, params,
+        host_allocator, out_device);
+  }
```

Comment:
device indices are backend specific and don't make sense here - and this isn't used - best to just error out on unimplemented